test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/AddressRegistrationTests.cs

```diff
@@ -40,58 +56,110 @@ public class AddressRegistrationTests
 
             var hostBuilder = new WebHostBuilder()
                 .UseConfiguration(config)
-                .UseKestrel()
+                .UseKestrel(options =>
+                {
+                    options.UseHttps(@"TestResources/testCert.pfx", "testPassword");
+                })
                 .Configure(ConfigureEchoAddress);
 
             using (var host = hostBuilder.Build())
             {
                 host.Start();
 
-                using (var client = new HttpClient())
+                RemoteCertificateValidationCallback validationCallback =
+                        (sender, cert, chain, sslPolicyErrors) => true;
+
+                try
                 {
-                    foreach (var testUrl in testUrls)
+#if NET451
+                    var handler = new HttpClientHandler();
+                    ServicePointManager.ServerCertificateValidationCallback += validationCallback;
+#else
+                    var handler = new WinHttpHandler();
```

Comment:
ServerCertificateValidationCallback was recently added to CurlHandler (see https://github.com/dotnet/corefx/pull/7060).    I will attempt to modify our tests to use CurlHandler on Linux and Mac OS.